libbpf-cargo/src/btf/btf.rs

```diff
@@ -355,6 +355,24 @@ impl<'a> Btf<'a> {
     ///
     /// `ty` must be a struct, union, enum, or datasec type.
     pub fn type_definition(&self, type_id: u32) -> Result<String> {
+        let next_type = |mut id| -> Result<Option<u32>> {
+            loop {
+                match self.type_by_id(id)? {
+                    BtfType::Struct(_)
+                    | BtfType::Union(_)
+                    | BtfType::Enum(_)
+                    | BtfType::Datasec(_) => return Ok(Some(id)),
+                    BtfType::Ptr(t) => id = t.pointee_type,
+                    BtfType::Array(t) => id = t.val_type_id,
+                    BtfType::Volatile(t) => id = t.type_id,
+                    BtfType::Const(t) => id = t.type_id,
+                    BtfType::Restrict(t) => id = t.type_id,
+                    BtfType::Typedef(t) => id = t.type_id,
+                    _ => return Ok(None),
+                }
+            }
+        };
+
         let is_terminal = |id| -> Result<bool> {
```

Comment:
is is_terminal still being used? If yes, clearly Array is not terminal, right? So need fixing as well?

Reply:
That's fine, let's leave it as is, thanks.